syspage.c

```diff
@@ -239,26 +244,40 @@ void *syspage_getAddress(void)
 
 /* General functions */
 
-void syspage_save(void)
+int syspage_save(void)
 {
 	int i;
 
+	if (syspage_common.syspage == NULL)
+		return ERR_MEM;
+
+	/* Save syspage arguments */
 	syspage_common.argCnt++; /* The last char is '\0' */
 	syspage_common.syspage->arg = (char *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	syspage_common.syspage->syspagesz += syspage_common.argCnt;
 	low_memcpy((void *)(syspage_common.syspage->arg), syspage_common.args, syspage_common.argCnt);
 
+	/* Save programs */
 	syspage_common.syspage->progs = (syspage_program_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	syspage_common.syspage->progssz = syspage_common.progsCnt;
 	syspage_common.syspage->syspagesz += (syspage_common.progsCnt * sizeof(syspage_program_t));
 	low_memcpy((void *)(syspage_common.syspage->progs), syspage_common.progs, syspage_common.progsCnt * sizeof(syspage_program_t));
 
+	/* Save maps */
 	syspage_common.syspage->maps = (syspage_map_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	for (i = 0; i < syspage_common.mapsCnt; ++i)
 		low_memcpy((void *)((void *)syspage_common.syspage->maps + i * sizeof(syspage_map_t)), &syspage_common.maps[i].map, sizeof(syspage_map_t));
 
 	syspage_common.syspage->syspagesz += (syspage_common.mapsCnt * sizeof(syspage_map_t));
 	syspage_common.syspage->mapssz = syspage_common.mapsCnt;
+
+	/* Save architecture dependent structure */
+	if (sizeof(syspage_arch_t) != 0) {
```

Comment:
I don't think we need this `if` really. `memcpy` of zero size should result in no operation anyway.

Reply:
I wanted to use `if` statement to make this operation more readable to inform that this structure is not always a part of syspage.
If it is clear without `if`, I can remove it.